Load form if active

jgoley opened the pull request:
- Loads forms and questions when navigating to the forms page
- sorts and filters the forms (it may be loading up the forms in the wrong order based on whether a form has been triggered and pending—I'll test this further)
- the async stuff needs to be refactored - not my specialty.
- I also need to fix acceptance routing - for some reason the user is no longer redirected to the first form.

jgoley committed:
Fetch forms on details page, not in list view

jgoley committed:
Load, filter and sort forms on forms page

jgoley committed:
Load questions if not cached

jgoley committed:
Show message on form page if no active forms

dan-nyanko commented:
@jgoley Acceptance routing is not working as you noted.  Also, if I accept a survey, then press back, then I'm stuck inside an infinite loading state.

jgoley committed:
Fix accept/reject routing and refactor

jgoley committed:
Remove form type buttons from accepted survey details page

jgoley committed:
Add message notifying user of count of remaining forms

jgoley commented:
@dan-nyanko I fixed the issue with acceptance routing and, I think, the infinite loop issue (I can't replicate it now).
I also added a little message letting the user know there are no active forms and how many 'future forms' are remaining (We should eventually add a button under that message to take the user back to surveys for a clean UX).

~~We may want to re-cache the forms because I think the pending forms may not update when they go active.~~

nathanathan commented on js/views/FormPage.js:
```diff
@@ -135,6 +162,49 @@ const FormPage = React.createClass ({
+  },
+
+  triggers(form, self, callback){
+    loadTriggers(form, self.props.survey, function(err, triggers){
```
Why are triggers being loaded from Parse rather than from the realm cache?

jgoley replied:
Good catch...I've switched to using the cached triggers. It certainly simplifies things a good bit. Thanks, @nathanathan.

jgoley committed:
Get cached triggers instead of fetching from Parse server

jgoley committed:
Refactor - ensure variables are declared

dan-nyanko commented on js/views/SurveyDetailsPage.js:
```diff
@@ -81,7 +111,6 @@ const SurveyDetailsPage = React.createClass ({
-    // this.props.navigator.push({path: 'surveylist', title: 'Surveys', sceneConfig: 'FloatFromLeft'})
   },
 
   showForms(tab) {
```
this method doesn't do anything, maybe we should just remove

jgoley replied:
Will do. Should we keep the code related to the form type menu? At this point it isn't used because we only have one tigger type and once a survey is selected, the user is directed to the first active form.

dan-nyanko replied:
@jgoley renderFormButtons method isn't used as well due to having a single trigger type, I'll add a note in pivotal so we don't forget this.  IMO its confusing leaving so much dead code in place.
